openapi3/swagger_loader.go

```diff
@@ -58,6 +59,38 @@ func (swaggerLoader *SwaggerLoader) loadSwaggerFromURIInternal(location *url.URL
 	return swaggerLoader.loadSwaggerFromDataWithPathInternal(data, location)
 }
 
+// loadSingleElementFromURI read the data from ref and unmarshal to JSON to the
+// passed element.
+func (swaggerLoader *SwaggerLoader) loadSingleElementFromURI(ref string, rootPath *url.URL, element json.Unmarshaler) error {
+	if !swaggerLoader.IsExternalRefsAllowed {
+		return fmt.Errorf("Encountered non-allowed external reference: '%s'", ref)
+	}
+
+	parsedURL, err := url.Parse(ref)
+	if err != nil {
+		return err
+	}
+
+	if parsedURL.Fragment != "" {
+		panic("References to files which contains more than one element definition are not supported")
```

Comment:
Can't this be an error?

Reply:
I left it that way because I thought if we get to that section of the code then it is clearly a programmer error but I guess panicking is almost never okay? It is now changed to an error.